testing/test_doctest.py

```diff
@@ -1224,3 +1226,19 @@ class Example(object):
     )
     result = testdir.runpytest("--doctest-modules")
     result.stdout.fnmatch_lines(["* 1 passed *"])
+
+
+class Broken:
+    def __getattr__(self, _):
+        raise KeyError("This should be an AttributeError")
+
+
+@pytest.mark.parametrize(
+    "stop", [None, lambda f: None, lambda f: False, lambda f: True]
+)
+def test_warning_on_unwrap_of_broken_object(stop):
+    bad_instance = Broken()
+    with _patch_unwrap_mock_aware():
+        with pytest.raises(KeyError):
+            with pytest.warns(pytest.PytestWarning):
+                inspect.unwrap(bad_instance, stop=stop)
```

Comment:
Maybe something is causing coverage tracing to be reset here - like it happens with RuntimeErrors.